src/plugins/workspace/public/register_default_collaborator_types.tsx

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { i18n } from '@osd/i18n';
+import { CoreSetup, OverlayRef } from '../../../core/public';
+import { toMountPoint } from '../../../plugins/opensearch_dashboards_react/public';
+
+import { WorkspaceCollaboratorTypesService, WorkspaceCollaboratorType } from './services';
+import {
+  AddCollaboratorsModal,
+  AddCollaboratorsModalProps,
+} from './components/add_collaborators_modal';
+
+export const generateOnAddCallback: (
+  options: Omit<AddCollaboratorsModalProps, 'onClose' | 'onAddCollaborators'> & {
+    getStartServices: CoreSetup['getStartServices'];
+  }
+) => WorkspaceCollaboratorType['onAdd'] = ({ getStartServices, ...props }) => async ({
+  onAddCollaborators,
+}) => {
+  let overlayRef: OverlayRef | null = null;
+  const [coreStart] = await getStartServices();
+  overlayRef = coreStart.overlays.openModal(
+    toMountPoint(
+      <AddCollaboratorsModal
+        {...props}
+        onClose={() => {
+          overlayRef?.close();
+        }}
+        onAddCollaborators={async (collaborators) => {
+          await onAddCollaborators(collaborators);
```

Comment:
Shall we try/catch here? perhaps could be a future improvement

Reply:
What should we do after catch an error? Do we need to prevent the modal close? Add a try catch would be better, but it's up to the collaborator type register. I think we can added it when we need it.